src/vendor/vendor.c

```diff
@@ -1792,12 +1772,7 @@ int DiscoverDM_ForAllComponents(char *objs_filename, char *params_filename)
     USP_ARG_Init(&rdk_params);
     USP_ARG_Init(&rdk_missing_objs);
 
-    // Iterate over all RDK components, discovering the parameters and objects that they provide
-    for (i=0; i<num_rdk_components; i++)
-    {
-        rdkc = &rdk_components[i];
-        (void)DiscoverDM_ForComponent(rdkc, &rdk_objects, &rdk_params);  // Intentionally ignoring any errors
-    }
+    DiscoverDM_ForComponent(&rdk_objects, &rdk_params);  // Intentionally ignoring any errors
 
     // Determine all table objects which do not currently have any instances, but do have a 'NumberOfEntries' parameter
     AddMissingObjs(&rdk_objects, &rdk_params, &rdk_missing_objs);
```

Comment:
I don't think AddMissingObjs() is needed anymore. It was necessary with CCSP API because there was no real way to discover the data model other than performing a get of Device. And that method is quite error prone if table objects do not have any instances in them. AddMissingObjs() was there to try and discover these missing table objects.
rbusElementInfo_get() doesn't suffer from the same issue.